compliance/compliance.go

```diff
@@ -2002,3 +2030,114 @@ func ForwardReferencesDisallowed(c *fluent.GRIBIClient, t testing.TB, _ ...TestO
+		},
+		func() {
+			c.Modify().AddEntry(t, fluent.NextHopEntry().WithNetworkInstance(defaultNetworkInstanceName).WithIndex(2).WithIPAddress("203.0.113.1"))
+			c.Modify().AddEntry(t, fluent.NextHopGroupEntry().WithNetworkInstance(defaultNetworkInstanceName).WithID(12).AddNextHop(2, 1))
+			c.Modify().AddEntry(t, fluent.IPv4Entry().WithPrefix("198.51.100.0/24").WithNetworkInstance(vrfName).WithNextHopGroup(12).WithNextHopGroupNetworkInstance(defaultNetworkInstanceName))
+		},
+	}
+
+	res := DoModifyOps(c, t, ops, fluent.InstalledInFIB, false)
+
+	chk.HasResult(t, res,
+		fluent.OperationResult().
+			WithIPv4Operation("203.0.113.1/32").
+			WithOperationType(constants.Add).
+			WithProgrammingResult(fluent.ProgrammingFailed).
+			AsResult(),
+		chk.IgnoreOperationID())
+}
+
+// IPv4EntryEmptyNextHopGroup - Empty NextHopGroup for the IPv4Entry 203.0.113.1/32.
+func IPv4EntryEmptyNextHopGroup(c *fluent.GRIBIClient, t testing.TB, _ ...TestOpt) {
+	defer flushServer(c, t)
+
+	ops := []func(){
+		func() {
+			c.Modify().AddEntry(t, fluent.NextHopEntry().WithNetworkInstance(defaultNetworkInstanceName).WithIndex(1).WithIPAddress("192.0.2.3"))
+			c.Modify().AddEntry(t, fluent.NextHopGroupEntry().WithNetworkInstance(defaultNetworkInstanceName).WithID(11))
+			c.Modify().AddEntry(t, fluent.IPv4Entry().WithPrefix("203.0.113.1/32").WithNetworkInstance(defaultNetworkInstanceName).WithNextHopGroup(11))
+		},
+		func() {
+			c.Modify().AddEntry(t, fluent.NextHopEntry().WithNetworkInstance(defaultNetworkInstanceName).WithIndex(2).WithIPAddress("203.0.113.1"))
+			c.Modify().AddEntry(t, fluent.NextHopGroupEntry().WithNetworkInstance(defaultNetworkInstanceName).WithID(12).AddNextHop(2, 1))
+			c.Modify().AddEntry(t, fluent.IPv4Entry().WithPrefix("198.51.100.0/24").WithNetworkInstance(vrfName).WithNextHopGroup(12).WithNextHopGroupNetworkInstance(defaultNetworkInstanceName))
+		},
+	}
+
+	res := DoModifyOps(c, t, ops, fluent.InstalledInFIB, false)
+
+	chk.HasResult(t, res,
+		fluent.OperationResult().
+			WithIPv4Operation("203.0.113.1/32").
+			WithOperationType(constants.Add).
+			WithProgrammingResult(fluent.ProgrammingFailed).
+			AsResult(),
+		chk.IgnoreOperationID())
+}
+
+// IPv4EntryEmptyNextHop - Empty NextHop for the IPv4Entry 203.0.113.1/32.
+func IPv4EntryEmptyNextHop(c *fluent.GRIBIClient, t testing.TB, _ ...TestOpt) {
+	defer flushServer(c, t)
+
+	ops := []func(){
+		func() {
+			c.Modify().AddEntry(t, fluent.NextHopEntry().WithNetworkInstance(defaultNetworkInstanceName).WithIndex(1))
+			c.Modify().AddEntry(t, fluent.NextHopGroupEntry().WithNetworkInstance(defaultNetworkInstanceName).WithID(11).AddNextHop(1, 1))
+			c.Modify().AddEntry(t, fluent.IPv4Entry().WithPrefix("203.0.113.1/32").WithNetworkInstance(defaultNetworkInstanceName).WithNextHopGroup(11))
+		},
+		func() {
+			c.Modify().AddEntry(t, fluent.NextHopEntry().WithNetworkInstance(defaultNetworkInstanceName).WithIndex(2).WithIPAddress("203.0.113.1"))
```

Comment:
Ditto to the comment above -- if we expect that these entries succeed, then let's check that their response code indicates as such.